src/libraries/Common/src/System/Net/IPv4AddressHelper.Common.cs

```diff
@@ -14,10 +14,24 @@ internal static partial class IPv4AddressHelper
         private const int NumberOfLabels = 4;
 
         // Only called from the IPv6Helper, only parse the canonical format
-        internal static unsafe int ParseHostNumber(ReadOnlySpan<char> str, int start, int end)
+        internal static int ParseHostNumber(ReadOnlySpan<char> str, int start, int end)
         {
-            byte* numbers = stackalloc byte[NumberOfLabels];
-            ParseCanonical(str, numbers, start, end);
+            Span<byte> numbers = stackalloc byte[NumberOfLabels];
+
+            for (int i = 0; i < numbers.Length; ++i)
+            {
+                int b = 0;
+                char ch;
+
+                for (; (start < end) && (ch = str[start]) != '.' && ch != ':'; ++start)
+                {
+                    b = (b * 10) + ch - '0';
+                }
+
+                numbers[i] = (byte)b;
+                ++start;
+            }
+
             return (numbers[0] << 24) + (numbers[1] << 16) + (numbers[2] << 8) + numbers[3];
```

Comment:
Can this be changed to `return BinaryPrimitives.ReadInt32BigEndian(numbers);`? I'm working on a separate PR for addressing this type of code pattern where I see it, but might be easier to address it in this PR so I don't introduce a conflict on my side.

Reply:
Sure. Thanks.